### PR TITLE
Feature-complete Auth component

### PR DESCRIPTION
**Pull request by peterszerzo**
This PR implements the API proposed in the previous Auth PR, which only handled input state. Status:
* [x] processing and error states
* [x] styling improvements 
* [x] documentation
* [x] review

(documentation is a bit sparse, pending component changes as we use it in production)

### CODE REVIEW

File: packages/blocks/src/Auth.tsx [hunk 1]
```diff
@@ -29,80 +29,117 @@ const Container = glamorous.div(({ theme }: { theme: Theme }): {} => ({
...
+              <TitleType css={{ textAlign: "center", margin: 0 }}>{this.props.title}</TitleType>
+            ) : null}
+            {this.props.error ? <ErrorNotice>{this.props.error}</ErrorNotice> : null}
+            <InputFields>
+              {this.props.username ? (
+                <Input
+                  css={inputStyle}
+                  value={this.props.username}
+                  label="User name"
+                  onChange={(v: string) => {
```
Comment (TejasQ) [inline packages/blocks/src/Auth.tsx]: Consider moving this function elsewhere and referencing it here for readability.

Comment (peterszerzo) [reply packages/blocks/src/Auth.tsx]: I don't think isn't worth extracting and making the file look more scattered than it already is. Not for 6-line functions that do something so repetitive and predictable.

Comment (TejasQ) [reply packages/blocks/src/Auth.tsx]: Okay.

File: packages/blocks/src/Auth.tsx [hunk 2]
```diff
@@ -68,8 +69,11 @@ const inputStyle: {} = {
   margin: "20px 0"
 }
 
+const exists = (s?: string): boolean => !!s || s === ""
```
Comment (TejasQ) [inline packages/blocks/src/Auth.tsx]: `s`? Please keep this a little less magical to save developers of mental overhead.

File: packages/blocks/src/Auth.tsx [hunk 3]
```diff
@@ -68,8 +69,11 @@ const inputStyle: {} = {
...
+
 export default class Auth extends React.Component<IProps, IState> {
   render() {
+    console.log("is username", !!this.props.username)
```
Comment (TejasQ) [inline packages/blocks/src/Auth.tsx]: no

Comment (peterszerzo) [reply packages/blocks/src/Auth.tsx]: Appreciate the catch!

File: packages/blocks/src/Auth.tsx (hunk 2 above)
Comment (peterszerzo) [reply packages/blocks/src/Auth.tsx]: Will do

File: packages/blocks/src/Auth.tsx [hunk 4]
```diff
@@ -69,11 +69,10 @@ const inputStyle: {} = {
   margin: "20px 0"
 }
 
-const exists = (s?: string): boolean => !!s || s === ""
+const exists = (stringValue?: string): boolean => !!stringValue || stringValue === ""
```
Comment (TejasQ) [inline packages/blocks/src/Auth.tsx]: Sorry to be a pain, but can you add a tiny comment explaining the intent of this function?

Comment (peterszerzo) [reply packages/blocks/src/Auth.tsx]: Nooo, I think that's a very good suggestion!

File: packages/blocks/src/Auth.tsx [hunk 5]
```diff
@@ -69,7 +69,14 @@ const inputStyle: {} = {
...
 
-const exists = (stringValue?: string): boolean => !!stringValue || stringValue === ""
+/*
+ * Checks whether a string value exists or not (same as !!stringValue, except it also returns
+ * true for empty string. This is used especially often in this component because this component 
+ * renders form fields if any corresponding string value is passed down in props.
+ * e.g. props.username === null -> rendern no user name form field at all
+ *      props.username === "" -> render empty form field for user name
+ */
+const isStringValue = (stringValue?: string): boolean => !!stringValue || stringValue === ""
```
Comment (TejasQ) [inline packages/blocks/src/Auth.tsx]: Ah, incredible 👌

Comment (micha-f) [reply packages/blocks/src/Auth.tsx]: Typo "rendern"

Comment (peterszerzo) [reply packages/blocks/src/Auth.tsx]: True story.. correct in German these days though, oder? :)

**Pull request merged**